lib/nostr.js

```diff
@@ -17,155 +15,148 @@ export const DEFAULT_CROSSPOSTING_RELAYS = [
-    await relay.waitUntilConnected({ timeout })
-    return relay
-  }
-
-  get connected () {
-    return this.ws.readyState === WebSocket.OPEN
+export const RELAYS_BLACKLIST = []
+
+/**
+ * @import {NDKSigner} from '@nostr-dev-kit/ndk'
+ * @import { NDK } from '@nostr-dev-kit/ndk'
+ * @import {NDKNwc} from '@nostr-dev-kit/ndk'
+ * @typedef {Object} Nostr
+ * @property {NDK} ndk
+ * @property {function(string, {logger: Object}): Promise<NDKNwc>} nwc
+ * @property {function(Object, {privKey: string, signer: NDKSigner}): Promise<NDKEvent>} sign
+ * @property {function(Object, {relays: Array<string>, privKey: string, signer: NDKSigner}): Promise<NDKEvent>} publish
+ */
+export class Nostr {
+  /**
+   * @type {NDK}
+   */
+  _ndk = null
+
+  constructor ({ privKey, defaultSigner, relays, supportNip07 = false, ...ndkOptions } = {}) {
+    this._ndk = new NDK({
+      explicitRelayUrls: relays,
+      blacklistRelayUrls: RELAYS_BLACKLIST,
+      autoConnectUserRelays: false,
+      autoFetchUserMutelist: false,
+      clientName: 'stacker.news',
+      signer: defaultSigner ?? this.getSigner({ privKey, supportNip07 }),
+      ...ndkOptions
+    })
   }
 
-  get closed () {
-    return this.ws.readyState === WebSocket.CLOSING || this.ws.readyState === WebSocket.CLOSED
+  /**
+   * @type {NDK}
+   */
+  get ndk () {
+    return this._ndk
   }
 
-  async waitUntilConnected ({ timeout } = {}) {
-    let interval
-
-    const checkPromise = new Promise((resolve, reject) => {
-      interval = setInterval(() => {
-        if (this.connected) {
-          resolve()
-        }
-        if (this.closed) {
-          reject(new Error(`failed to connect to ${this.url}: ` + this.error))
-        }
-      }, 100)
-    })
-
-    try {
-      return await withTimeout(checkPromise, timeout)
-    } catch (err) {
-      this.close()
-      throw err
-    } finally {
-      clearInterval(interval)
-    }
+  /**
+   *
+   * @param {args} param0
+   * @param {string} [args.privKey] - private key to use for signing
+   * @param {boolean} [args.supportNip07] - whether to use NIP-07 signer if available
+   * @returns {NDKPrivateKeySigner | NDKNip46Signer | NDKNip07Signer | null} - a signer instance
+   */
+  getSigner ({ privKey, supportNip07 = true } = {}) {
+    if (privKey) return new NDKPrivateKeySigner(privKey)
+    if (supportNip07 && typeof window !== 'undefined' && window?.nostr) return new NDKNip07Signer()
+    return null
   }
 
-  close () {
-    const state = this.ws.readyState
-    if (state !== WebSocket.CLOSING && state !== WebSocket.CLOSED) {
-      this.ws.close()
-    }
+  /**
+   * @param {Object} rawEvent
+   * @param {number} rawEvent.kind
+   * @param {number} rawEvent.created_at
+   * @param {string} rawEvent.content
+   * @param {Array<Array<string>>} rawEvent.tags
+   * @param {Object} context
+   * @param {string} context.privKey
+   * @param {NDKSigner} context.signer
+   * @returns {Promise<NDKEvent>}
+   */
+  /* eslint-disable camelcase */
```

Comment:
This disables the rule for the remainder of the file. I think that's okay, so you can just put this line once somewhere at the top of the file.